course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,99 @@
+#include "unique_ptr.h"
+#include "unique_ptr.cc"
```

Comment:
This is why template classes don't have the .cc file, and are either entirely implementd in the .h. or - like in - your case, in the header files: one containing the class definition, and the other method definitions.

Different naming convetion are used for the second file: XXX_impl.h, XXX.hxx, XXX.ipp